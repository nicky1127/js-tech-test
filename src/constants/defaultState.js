const state = {
  liveEvents: [],
  event: {},
  markets: {},
  isOddsDecimal: false,
  loading: false,
  error: {},
  selections: []
};

export default state;
