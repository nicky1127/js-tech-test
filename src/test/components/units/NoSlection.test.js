import React from "react";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import NoSeleciotn from "components/units/NoSelection";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import defaultState from "constants/defaultState"; 

beforeEach(() => {
  jest.clearAllMocks();
  cleanup();
});

const props = {};

const state = {
  ...defaultState
};
const mockStore = configureMockStore();
const store = mockStore(state);

describe("NoSeleciotn", () => {
  it("Snapshot matchs", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <NoSeleciotn {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("NoSeleciotn should render one [data-testid=noSelection]", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <NoSeleciotn {...props} />
      </Provider>
    );
    const container = getByTestId("noSelection");  
    expect(container).toBeInTheDocument();
  });
});
