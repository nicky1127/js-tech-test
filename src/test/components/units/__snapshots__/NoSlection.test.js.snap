// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoSeleciotn Snapshot matchs 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="noSelection"
>
  <p
    className="MuiTypography-root makeStyles-noSelectText-2 MuiTypography-body1"
  >
    You have no selections in your Bet Slip
  </p>
  <div
    className="MuiBox-root MuiBox-root-4 makeStyles-optionBox-3"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Go to In-Play bets
    </p>
  </div>
  <div
    className="MuiBox-root MuiBox-root-5 makeStyles-optionBox-3"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Build an Accumulator
    </p>
  </div>
  <div
    className="MuiBox-root MuiBox-root-6 makeStyles-optionBox-3"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Next Races
    </p>
  </div>
</div>
`;
